crates/core/src/workflows/application.rs

```diff
@@ -66,8 +68,23 @@ pub async fn initialize(
     initialize_network(&instance_context).await?;
 
     // 4. (first initialization only) Call the init callbacks in the zomes
+    //    and publish the headers of the initial chain entries
     if first_initialization {
         call_init(dna, &instance_context).await?;
+
+        log_debug!(
+            context,
+            "dna/initialize: publishing headers of initial chain entries...",
+        );
+
+        for header in context.state().unwrap().agent().iter_chain() {
+            publish_header_entry(header.entry_address().clone(), &context).await?;
```

Comment:
Just for the record, did I understand this correct: you're adding this explicit header publish here because it makes scenario tests of some hApps work, which otherwise are flaky?

Reply:
@lucksus yes, I'm pretty sure that the rule we have in the consistency model for waiting for chain initialization is not working properly, so any scenario that relies on AgentID is not actually able to reliably wait for that entry to be held.

The proof is that without this branch, the telephone games which fetch AgentID as the check step fail without a hard timeout, but with this branch, they pass without a hard timeout, relying only on the consistency model.

Reply:
Hmm it is a shame to have to add this back at least we know we are not relying on it for things to work (the holding list does the actual sharing), but it is only for tests